src/game.zig

```diff
@@ -532,10 +532,58 @@ fn update(flecs_world: *flecs.World, gfx_state: *gfx.D3D12State) void {
         environment_info.world_time = world_time;
     }
 
-    gfx.update(gfx_state);
-    // gui_system.preUpdate(&gui_sys);
+    gfx.beginFrame(gfx_state);
 
     flecs_world.progress(dt);
-    // gui_system.update(&gui_sys);
-    gfx.draw(gfx_state);
+
+    const camera_comps = getActiveCamera(flecs_world);
+    gfx.endFrame(gfx_state, camera_comps.camera, camera_comps.transform.getPos00());
+}
+
+fn getActiveCamera(flecs_world: *flecs.World) struct { camera: *const fd.Camera, transform: *const fd.Transform } {
+    var query_builder_camera = flecs.QueryBuilder.init(flecs_world.*);
+    _ = query_builder_camera
+        .withReadonly(fd.Camera)
+        .withReadonly(fd.Transform);
+
+    var query_camera = query_builder_camera.buildQuery();
```

Comment:
this should probably be buildFilter. Filters are better for "on the fly" queries, Queries are better for stored queries.

Reply:
I've tried replacing this with a filter, but I'm hitting breakpoints in flecs. Can we change this together later?

Reply:
Yep!